### PR TITLE
feat: update the Kotlin release information and evolution principles

### PR DESCRIPTION
**Pull request by koshachy**
[KT-67865](https://youtrack.jetbrains.com/issue/KT-67865) [Docs][K2] update Kotlin Release Page
[KT-67944](https://youtrack.jetbrains.com/issue/KT-67944) [Docs][K2][IDE] Update IDE support description for K2

### CODE REVIEW

File: docs/topics/releases.md [hunk 1]
```diff
@@ -1,30 +1,57 @@
...
-* _Bug fix releases_ (1._x_._yz_) that include bug fixes for incremental releases.
+* _Language releases_ (2._x_._0_) that bring major changes in the language. Released once in 6 months.
+* _Tooling releases_ (2._x_._20_) that are shipped between language releases and include updates in the tooling,
+    performance improvements, and bug fixes.
+    Released in 3 months after corresponding _language release_.
+* _Bug fix releases_ (2._x_._yz_) that include bug fixes for _tooling releases_. There is no exact release schedule for these releases.
 
-For example, for the feature release 1.3 we had several incremental releases including 1.3.10, 1.3.20, and 1.3.70.
-For 1.3.70, we had 2 bug fix releases – 1.3.71 and 1.3.72.
+> For example, for the feature release 1.8.0, we had only one tooling release 1.8.20,
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/releases.md:11]: Yeah, a bit strange.
Let's comment this tip and bring it back when 2.1 happen.

File: docs/topics/kotlin-evolution.md [hunk 2]
```diff
@@ -32,120 +32,177 @@ want most of the code in the world to be already updated and thus have no issues
...
+All changes to the Kotlin Programming Language are overseen by the [Lead Language Designer](https://kotlinfoundation.org/structure/) (currently Michail Zarečenskij).
+The Lead Designer has the final say in all matters related to language evolution.
+Additionally, incompatible changes to fully stable components have to be approved to by the [Language Committee](https://kotlinfoundation.org/structure/) designated 
+under the [Kotlin Foundation](https://kotlinfoundation.org/structure/) (currently comprising Jeffrey van Gogh, Werner Dietl, and Michail Zarečenskij).
 
-The Language Committee makes final decisions on what incompatible changes will be made and what exact measures should be taken to make user updates comfortable. In doing so, it relies on a set of guidelines available [here](https://kotlinfoundation.org/language-committee-guidelines/).
+The Language Committee makes final decisions on what incompatible changes will be made and what exact measures should be taken to make user updates comfortable.
+In doing so, it relies on a set of [Language committee guidelines](https://kotlinfoundation.org/language-committee-guidelines/).
 
 ## Feature releases and incremental releases
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/kotlin-evolution.md]: Updated everywhere on a page

File: docs/topics/releases.md [hunk 3]
```diff
@@ -1,30 +1,57 @@
 [//]: # (title:  Kotlin releases)
 
-We ship different types of releases:
+Since Kotlin 2.0.0, we ship the following types of releases:
 
-* _Feature releases_ (1._x_) that bring major changes in the language.
-* _Incremental releases_ (1._x_._y_) that are shipped between feature releases and include updates in the tooling,
-  performance improvements, and bug fixes.
-* _Bug fix releases_ (1._x_._yz_) that include bug fixes for incremental releases.
+* _Language releases_ (2._x_._0_) that bring major changes in the language. Released once in 6 months.
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/releases.md]: Added

File: docs/topics/whatsnew14.md [hunk 4]
```diff
@@ -1517,7 +1517,7 @@ to help you decide which suggestions to accept and which to ignore.
 
 ![Migration inspections](migration-inspection-wn.png)
 
-Kotlin 1.4.0 is a [feature release](kotlin-evolution.md#feature-releases-and-incremental-releases) and therefore can 
+Kotlin 1.4.0 is a [feature release](kotlin-evolution.md#language-and-tooling-releases) and therefore can
```
Comment (koshachy) [inline docs/topics/whatsnew14.md:1520]: For the record, not sure that we should fix the namen for the previous 1.x.x release.
I fixed the anchors just to be sure that everything works.

**Review by woainikk — APPROVED** [review]

**Pull request merged**